added self query for vectara vector store

adeelehsan opened the pull request:
Add support for Self-query for the Vectara integration.

To test it please run the example in `examples/src/retrievers/vectara_self_query.ts`
Steps to run the example
    

> 1. add .env file in example directory and put the vectara credentials and openAI key 
> 2. Command to run the example `yarn example examples/src/retrievers/vectara_self_query.ts`

jacoblee93 committed:
Delete artifacts (#3305)

bracesproul committed:
Add missing docs (#3290)

* add wikipedia tool

* added searxng

* adderd SearchApi

* added pdf load dir

* add cohere example

* chore: lint files

* update examples to use runnable agents

bracesproul committed:
Brace/new api refs build (#3303)

* Update api refs build script

* update scripts

* update scripts, remove unused files

* rm langchain

* cleanup scripts

* gave better name

* remove echo

* Update package.json

* revert cjs

* cr

* rm unused script

* cr

dqbd committed:
Fix broken fetch usage for CFW (#3302)

Fixes #3187

jacoblee93 committed:
Bump Anthropic + OpenAI versions (#3308)

* Bump OpenAI version

* Bump Anthropic SDK

* Update test

* Whoops

jacoblee93 committed:
Hotfix pdf (#3306)

* fix: remove blank space between words when use pdf-loader

* fix: add skipBlank option in pdf-loader

* Change PDF loader instance method name, add docs

* Formatting

---------

Co-authored-by: ppxu <xyang0110@126.com>

Njuelle committed:
Add PrismaVectorStore filter IN operator (#3304)

* add Prisma filter IN operator

* better naming

* Update langchain/src/vectorstores/prisma.ts

Co-authored-by: Brace Sproul <braceasproul@gmail.com>

* use typeguard

* add operators doc

* add in operator in doc

* typo

* Update langchain/src/vectorstores/prisma.ts

Co-authored-by: Brace Sproul <braceasproul@gmail.com>

* Update examples/src/indexes/vector_stores/prisma_vectorstore/prisma.ts

Co-authored-by: Brace Sproul <braceasproul@gmail.com>

* prettier

* Update prisma.mdx

* Update prisma.mdx

---------

Co-authored-by: Brace Sproul <braceasproul@gmail.com>
Co-authored-by: Jacob Lee <jacoblee93@gmail.com>

omikader committed:
feat(apify): support Document[] return type for mapping function (#3262)

* feat(apify): support Document[] for mapping function

* misc: make it backwards compatible

kwadhwa18 committed:
Integrate Rockset as a vector store (#3231)

* Integrate Rockset as a vector store

* address comments

* Mark Rockset as requiring optional dep

* Fix lint

* Fix build

* Format

---------

Co-authored-by: jacoblee93 <jacoblee93@gmail.com>

vdeturckheim committed:
feat: add file-system based cache (#3089)

* feat: add file-system based cache

* feat: add file-system based cache

* feat: add file-system based cache

* Rename file_system.ts to file_system.test.ts

* Update file_system.test.ts

* Update file_system.ts

* Fix lint

* Fix build

* Adds docs

* Format

* Format

---------

Co-authored-by: Jacob Lee <jacoblee93@gmail.com>

jacoblee93 committed:
Release 0.0.192

jacoblee93 committed:
Merge pull request #3311 from langchain-ai/release

Release 0.0.192

Commit:
added self query for vectara vector store

Commit:
updated the docs

Commit:
skip the integration test

jacoblee93 committed:
Use .invoke for all agent docs and examples (#3319)

bracesproul committed:
[AUTO-GENERATED] Add JSDoc examples to classes. (#3309)

* [AUTO-GENERATED] Add JSDoc examples to classes.

* lint files

* chore: lint files

bracesproul committed:
updated langchain stack img to be svg (#3324)

bracesproul committed:
[AUTO-GENERATED] Add JSDoc examples to classes. (#3325)

* [AUTO-GENERATED] Add JSDoc examples to classes.

* [AUTO-GENERATED] Add JSDoc examples to classes.

* cr

* chore: lint files

bracesproul committed:
[AUTO-GENERATED] Add JSDoc examples to classes. (#3327)

* [AUTO-GENERATED] Add JSDoc examples to classes.

* cr

bracesproul committed:
[AUTO-GENERATED] Add JSDoc examples to classes. (#3329)

* [AUTO-GENERATED] Add JSDoc examples to classes.

* cr

* Update langchain/src/chains/constitutional_ai/constitutional_principle.ts

bracesproul committed:
[AUTO-GENERATED] Add JSDoc examples to classes. (#3330)

* [AUTO-GENERATED] Add JSDoc examples to classes.

* cr

ofermend commented on examples/src/retrievers/vectara_self_query.ts:
```diff
@@ -0,0 +1,132 @@
+  /**
+   * We need to create a basic translator that translates the queries into a
+   * filter format that the vector store can understand. We provide a basic translator
+   * translator here, but you can create your own translator by extending BaseTranslator
```
nit: the word "translator" appears twice.

ofermend commented on langchain/src/retrievers/self_query/tests/vectara_self_query.init.test.ts:
```diff
@@ -0,0 +1,113 @@
+import { FakeEmbeddings } from "../../../embeddings/fake.js";
+import { VectaraStore } from "../../../vectorstores/vectara.js";
+
+test.skip("Vectara Self Query Retriever Test", async () => {
```
Shouldn't this be test() and not test.skip()

adeelehsan replied:
Since this test requires the credentials and account setup of vectara thats why it is skipped. if we don't skip it would fail whenever test suit runs. Other vector stores are also using the `test.skip` .

ofermend replied:
Yes you are right. Good point.

ofermend commented on langchain/src/retrievers/self_query/tests/vectara_self_query.init.test.ts:
```diff
@@ -0,0 +1,113 @@
+    }),
+  ];
+
+  const attributeInfo: AttributeInfo[] = [
```
Perhaps add some comments to let people running the test know that they need to add these metadata fields to their Vectara corpus for this to work?

jacoblee93 committed:
Update Ollama functions (#3336)

raioalbano committed:
Remove console.log from googlevertexai-connection.ts (#3322)

* Remove console.log from googlevertexai-connection.ts

Remove console.log from googlevertexai-connection.ts

Problem:
- In case of invalid address, an error is logged exposing authentication token  in logs.
- this log should be handled by langchain callback mechanism.

* fix yarn lint

---------

Co-authored-by: Raissa Albano Fujioka <ralbanofujio@opentext.com>

hinthornw committed:
Add batch size arg (#3310)

faileon committed:
init (#3312)

netzhuffle committed:
Improve MessageContent type (#3318)

* Improve MessageContent type

The fields are mutually exclusive. This represents this in its type.

* Add auto option to image detail

Co-authored-by: Brace Sproul <braceasproul@gmail.com>

---------

Co-authored-by: Brace Sproul <braceasproul@gmail.com>

Njuelle committed:
add missing prisma filter operators (#3321)

jacoblee93 committed:
Release 0.0.193

jacoblee93 committed:
Merge pull request #3337 from langchain-ai/release

Release 0.0.193

Commit:
Updated the comments in the example

adeelehsan committed:
Merge branch 'main' into adding-self-query-for-vectara

jacoblee93 committed:
Rename test, add linter warning
